### PR TITLE
feat(logs): add loki logs

### PR DESCRIPTION
**Pull request by burgerni10**

### CODE REVIEW

**Comment by burgerni10** [comment]
Working poc to send data to loki : 

image

For now, I take the engineName, so if we set up loki logs, we need to rename the oibus engine so we can differentiate them.

To do : 
-basic auth
-send the logs to another OIBus
-get the logs from another OIBus and send them to loki (or to another OIBus ?)

File: src/engine/Logger.class.js [hunk 1]
```diff
@@ -31,16 +32,44 @@ class Logger {
...
+        fileName: sqliteLog.fileName,
+        level: sqliteLog.level,
+        maxFileSize: sqliteLog.maxSize,
+        handleExceptions: true,
+      }))
+    }
+
+    if (lokiLog.level !== 'none') {
+      this.logger.add(new LokiTransport({
+        host: lokiLog.host, // http://loki.igw.rd.optimistik.fr:3100
```
Comment (marouanehassanioptimistik) [inline src/engine/Logger.class.js:65]: please remove this as it will spread our loki url

**Review by marouanehassanioptimistik — CHANGES_REQUESTED** [review]

File: src/south/ProtocolHandler.class.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ class ProtocolHandler {
 
     const { logParameters } = this.dataSource
     this.logger = new Logger()
-    this.logger.changeParameters(this.engineConfig.logParameters, logParameters)
+    this.logger.changeParameters(this.engineConfig.engineName, this.engineConfig.logParameters, logParameters)
```
Comment (jfhenon) [inline src/south/ProtocolHandler.class.js:50]: maybe we should only send the engineConfig and it's up to the called function to pick engineName and logParameters.

**Review by jfhenon — COMMENTED** [review]
- Should the unique ID for Loki be a parameter at the Loki config level (making the OIBus engine name unique among all users may not be optimal).
- We don't solve the case of OIBus with no access to Internet. We should add the ability to forward Logs to another OIBus
- We may not need the OIBus alive anymore (or limit it to a regular log)

**Comment by burgerni10** [comment]
> * Should the unique ID for Loki be a parameter at the Loki config level (making the OIBus engine name unique among all users may not be optimal).
=> I don't see the point of the engineName anyway : should we juste have a 'loki-id' in the loki log level and remove the engine name from the config ?

> * We don't solve the case of OIBus with no access to Internet. We should add the ability to forward Logs to another OIBus
=> Will be done in a following PR

> * We may not need the OIBus alive anymore (or limit it to a regular log)
=> I agree that the Alive signal is OIAnalytics specific : people not using OIA must implement an endpoint for it. So for me it makes sense to remove the alive signal, once issue #1154 is done (it makes sense to do it now that winston-loki is almost implemented)

**Review by jfhenon — APPROVED** [review]

**Review by marouanehassanioptimistik — APPROVED** [review]

**Pull request merged**